ecommerce/auth/router.py

```diff
@@ -1,30 +1,45 @@
-from fastapi import APIRouter, Depends, HTTPException, status
-from .schemas import UserOut, UserAuth
+from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
+from fastapi import HTTPException, status, Depends, APIRouter
 
 
-from ecommerce.main import get_db
-from ecommerce.User.model import Users
-from ecommerce.main import create_user
+from app.database import get_db
+from user.model import Users
+from user.utils import create_user, read_user, get_current_user
+from user.password import verify_password
+from auth.schemas import UserOut, UserAuth, TokenOut
+from auth.json_token import create_access_token, create_refresh_token
 
-router = APIRouter(tags=["auth"])
+router = APIRouter(tags=["Auth"])
 
 
-@router.post('/User/login')
-def login():
-    pass
+@router.post('/User/sing_up/', response_model=UserOut)
+async def register_user(data: UserAuth, database=Depends(get_db)):
+    user = read_user(data.email, database=database)
+    if user is not None:

```

Comment:
```suggestion
    if (user := read_user(data.email, database=database)) is not None:
```
There exists syntax called walrus operator `:=` that allows you assigning to variables inside `if statement`, thus reducing amount of code.